### PR TITLE
Convert ADD COLUMN SQL into pgroll operations

### PR DESCRIPTION
**Pull request by ryanslade**
Converts `ALTER TABLE ADD COLUMN` statements into `pgroll` operations. There are many combinations, at the moment we support the following:

```sql
ALTER TABLE foo ADD COLUMN bar int
ALTER TABLE foo ADD COLUMN bar int NOT NULL
ALTER TABLE schema.foo ADD COLUMN bar int
ALTER TABLE foo ADD COLUMN bar int DEFAULT 123
ALTER TABLE foo ADD COLUMN bar int DEFAULT 'baz'
ALTER TABLE foo ADD COLUMN bar int DEFAULT null
ALTER TABLE foo ADD COLUMN bar int NULL
ALTER TABLE foo ADD COLUMN bar int UNIQUE
ALTER TABLE foo ADD COLUMN bar int UNIQUE NOT DEFERRABLE
ALTER TABLE foo ADD COLUMN bar int UNIQUE INITIALLY IMMEDIATE
ALTER TABLE foo ADD COLUMN bar int PRIMARY KEY
ALTER TABLE foo ADD COLUMN bar int CHECK (bar > 0)
ALTER TABLE foo ADD COLUMN bar int CONSTRAINT check_bar CHECK (bar > 0)
ALTER TABLE foo ADD COLUMN bar int CONSTRAINT fk_baz REFERENCES baz (bar)
ALTER TABLE foo ADD COLUMN bar int CONSTRAINT fk_baz REFERENCES baz (bar) ON UPDATE NO ACTION
ALTER TABLE foo ADD COLUMN bar int CONSTRAINT fk_baz REFERENCES baz (bar) ON DELETE NO ACTION
ALTER TABLE foo ADD COLUMN bar int CONSTRAINT fk_baz REFERENCES baz (bar) ON DELETE RESTRICT
ALTER TABLE foo ADD COLUMN bar int CONSTRAINT fk_baz REFERENCES baz (bar) ON DELETE SET NULL 
ALTER TABLE foo ADD COLUMN bar int CONSTRAINT fk_baz REFERENCES baz (bar) ON DELETE SET DEFAULT
ALTER TABLE foo ADD COLUMN bar int CONSTRAINT fk_baz REFERENCES baz (bar) ON DELETE CASCADE
```

And fall back to raw SQL for these cases:

```sql
ALTER TABLE foo ADD COLUMN bar int REFERENCES bar (c) ON UPDATE RESTRICT
ALTER TABLE foo ADD COLUMN bar int REFERENCES bar (c) ON UPDATE CASCADE
ALTER TABLE foo ADD COLUMN bar int REFERENCES bar (c) ON UPDATE SET NULL
ALTER TABLE foo ADD COLUMN bar int REFERENCES bar (c) ON UPDATE SET DEFAULT
ALTER TABLE foo ADD COLUMN IF NOT EXISTS bar int
ALTER TABLE foo ADD COLUMN bar int UNIQUE DEFERRABLE
ALTER TABLE foo ADD COLUMN bar int UNIQUE INITIALLY DEFERRED
ALTER TABLE foo ADD COLUMN bar int GENERATED BY DEFAULT AS IDENTITY
ALTER TABLE foo ADD COLUMN bar int GENERATED ALWAYS AS ( 123 ) STORED
```

Part of https://github.com/xataio/pgroll/issues/504

### CODE REVIEW

File: pkg/sql2pgroll/alter_table.go [hunk 1]
```diff
@@ -380,6 +397,112 @@ func canConvertDropConstraint(cmd *pgq.AlterTableCmd) bool {
 	return cmd.Behavior != pgq.DropBehavior_DROP_CASCADE
 }
 
+func convertAlterTableAddColumn(stmt *pgq.AlterTableStmt, cmd *pgq.AlterTableCmd) (migrations.Operation, error) {
```
Comment (andrew-farries) [inline pkg/sql2pgroll/alter_table.go:400]: For consistency with other `convert*` functions this should have a comment too.

File: pkg/sql2pgroll/alter_table.go [hunk 2]
```diff
@@ -442,6 +565,24 @@ func getQualifiedRelationName(rel *pgq.RangeVar) string {
 	return fmt.Sprintf("%s.%s", rel.GetSchemaname(), rel.GetRelname())
 }
 
+// canConvertColumnDef returns true iff `col` can be converted to a pgroll `Column` definition.
+// TODO: Copied from WIP PR
+func canConvertColumnDef(col *pgq.ColumnDef) bool {
```
Comment (andrew-farries) [inline pkg/sql2pgroll/alter_table.go:570]: Another copy of this function was merged in #548. We should rebase this PR and use it.

File: pkg/sql2pgroll/alter_table.go [hunk 3]
```diff
@@ -380,6 +397,112 @@ func canConvertDropConstraint(cmd *pgq.AlterTableCmd) bool {
...
+	operation := &migrations.OpAddColumn{
+		Column: migrations.Column{
+			Name: columnDef.GetColname(),
+			Type: columnType,
+		},
+		Table: getQualifiedRelationName(stmt.GetRelation()),
+		Up:    PlaceHolderSQL,
+	}
+
+	// Currently only handles DEFAULT
```
Comment (andrew-farries) [inline pkg/sql2pgroll/alter_table.go:424]: I think this comment isn't accurate anymore right?

**Comment by ryanslade** [comment]
@andrew-farries Done, PTAL

**Pull request merged**